assets/src/edit-story/components/canvas/eyedropperLayer.js

```diff
@@ -0,0 +1,259 @@
+
+  useKeyDownEffect(document.body, 'esc', closeEyedropper);
+
+  if (!eyedropperActive || !img) {
+    return null;
+  }
+
+  const magnifierSize = 200; // We can use scroll wheel to control it?
+  const magnifierPixels = 7;
+  const magnifierRectSize = magnifierSize / magnifierPixels;
+
+  const magnify = function (x, y) {
+    const canvas = magnifier.current;
+    if (canvas) {
+      const ctx = canvas.getContext('2d');
+
+      // Sometimes initial setting is not enough.
+      ctx.imageSmoothingEnabled = false;
+      ctx.mozImageSmoothingEnabled = false;
+      ctx.webkitImageSmoothingEnabled = false;
+      ctx.msImageSmoothingEnabled = false;
+
+      // Draw enlarged cropped image.
+      ctx.clearRect(0, 0, magnifierSize, magnifierSize);
+      ctx.drawImage(
+        imgRef.current,
+        Math.round(x - magnifierPixels / 2),
+        Math.round(y - magnifierPixels / 2),
+        magnifierPixels,
+        magnifierPixels,
+        0,
+        0,
+        magnifierSize,
+        magnifierSize
+      );
+
+      // Draw center rectangle for better aiming.
+      ctx.beginPath();
+      ctx.rect(
+        Math.round(magnifierSize / 2) - magnifierRectSize / 2,
+        Math.round(magnifierSize / 2) - magnifierRectSize / 2,
+        magnifierRectSize,
+        magnifierRectSize
+      );
+      ctx.stroke();
+    }
+  };
+
+  const onMouseMove = (e) => {
+    const { left, top, width, height } =
+      fullbleedContainer.getBoundingClientRect();
+    const x = (e.clientX - left) * (pageWidth / width);
+    const y = (e.clientY - top) * (fullHeight / height);
+
+    if (x < 0 || y < 0 || x > width || y > height) {
+      return;
+    }
+
+    // Move magnifier canvas.
+    magnifierInfo.current.style.transform = `translate(${x}px, ${y}px)`;
+
+    // Redraw magnifier canvas.
+    magnify(x, y);
+
+    // Get and print pixel color.
+    const rgbaObject = getColorFromPixelData(eyedropperPixelData, x, y, width);
+    const { r, g, b, a } = rgbaObject;
+    const hex = rgba(r, g, b, a);
+    magnifierColor.current.style.background = `rgba(${r},${g},${b},${a})`;
+    magnifierColor.current.style.color = readableColor(
+      hex,
+      '#333',
+      '#EDEDED',
+      false
+    );
+    magnifierColor.current.innerText = hex;
+  };
+
+  const onClick = (e) => {
+    const { left, top, width, height } =
+      fullbleedContainer.getBoundingClientRect();
+    const x = (e.clientX - left) * (pageWidth / width);
+    const y = (e.clientY - top) * (fullHeight / height);
+
+    const rgbaObject = getColorFromPixelData(
+      eyedropperPixelData,
+      x,
+      y,
+      pageWidth
+    );
+    eyedropperCallback(rgbaObject);
+  };
+
+  return (
+    <EyedropperBackground>
+      <DisplayPageArea withSafezone={false} showOverflow>
+        {/* eslint-disable-next-line styled-components-a11y/click-events-have-key-events, styled-components-a11y/no-static-element-interactions */}
```

Comment:
Can you leave a comment about why we're ignoring the rules here for consistency (we're ideally adding a comment whenever disabling this rule)?